### PR TITLE
Fix validation for maxLatency and maxLatencyTier on Flow update.

### PR DESCRIPTION
**Pull request by dmitrii-beliakov**

### CODE REVIEW

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java [hunk 1]
```diff
@@ -288,7 +288,7 @@ private void checkFlowForMirrorEndpointConflicts(String flowId, EndpointDescript
             if ((flowMirrorPointsForward.isPresent() || flowMirrorPointsReverse.isPresent())
                     && (endpoint.isTrackLldpConnectedDevices() || endpoint.isTrackArpConnectedDevices())) {
                 String errorMessage = format("Flow mirror point is created for the flow %s, "
-                        + "lldp or arp can not be set to true.", flowId);
+                        + "LLDP or arp can not be set to true.", flowId);
```
Comment (niksv) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java:291]: minor: now we name LLDP in upper case and arp in lower case

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/ValidatorUtils.java [hunk 2]
```diff
@@ -25,21 +25,34 @@ public final class ValidatorUtils {
...
+            return;
+        }
+
+        if (maxLatency < 0) {
+            throw new InvalidFlowException("maxLatency cannot be negative", ErrorType.DATA_INVALID);
+        }
+        if (maxLatencyTier2 < 0) {
+            throw new InvalidFlowException("maxLatencyTier2 cannot be negative", ErrorType.DATA_INVALID);
+        }
+        if (maxLatency.equals(0L) && maxLatencyTier2.equals(0L)) {
```
Comment (niksv) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/ValidatorUtils.java:53]: if maxLatencyTier2 == 0 we will exit from method in line 44. So this `if` is always false

Comment (dmitrii-beliakov) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/ValidatorUtils.java:53]: Yeah, thanks! Somehow, I committed a wrong version, checkstyle is failing as well 🤷‍♂️. That block on lines 43-35 actually allows maxLatency to be negative, when tier2 is not set. I remember making it correct, but here we go again 😒

**Review by niksv — APPROVED** [review]

**Comment by DejfCold** [comment]
Removing `tested` and `ready for testing` labels as there will be some additional changes related to #5054

**Comment by DejfCold** [comment]
Fixes also #5054

Moved maxLatency-changed checks only from when the strategy changes to always so that the additional validations running on update (as opposed to on patch) are always ran.

**Review by niksv — APPROVED** [review]

**Comment by DejfCold** [comment]
For the sake of completion:
Closes #5054
Closes #5053

**Review by niksv — APPROVED** [review]

File: src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java [hunk 3]
```diff
@@ -342,13 +344,16 @@ private List<PathNodePayload> buildPathFromFlow(Flow flow, FlowPath flowPath) {
...
+    public Flow updateFlow(FlowOperationsCarrier carrier, FlowPatch flowPatch)
+            throws FlowNotFoundException, InvalidFlowException {
         String flowId = flowPatch.getFlowId();
         if (yFlowRepository.isSubFlow(flowId)) {
             throw new MessageException(ErrorType.REQUEST_INVALID, "Could not modify flow",
                     format("%s is a sub-flow of a y-flow. Operations on sub-flows are forbidden.", flowId));
         }
 
+        ValidatorUtils.validateMaxLatencyAndLatencyTier(flowPatch.getMaxLatency(), flowPatch.getMaxLatencyTier2());
+
```
Comment (niksv) [inline src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java:356]: This is not correct way to check latency in patch. 

Patch request can contain only fields which must be changed. For example patch request can have body 
```
{
    max_latecny_tire2: 123
}
```
Without max_latency.
If some field is missed - we must take field value from already existent flow. 
(look at lines 370-380) for example. Only after that we can compare latencies

**Review by niksv — CHANGES_REQUESTED** [review]

File: src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java [hunk 4]
```diff
@@ -458,6 +461,20 @@ private boolean updateRequiredBySource(FlowPatch flowPatch, Flow flow) {
...
+        }
+        return !Objects.equals(flowPatch.getMaxLatency(), flow.getMaxLatency());
+    }
+
+    private boolean updateRequiredByMaxLatencyTier2(FlowPatch flowPatch, Flow flow) {
+        if (flowPatch.getMaxLatencyTier2() == null) {
+            return false;
+        }
+        return !Objects.equals(flowPatch.getMaxLatencyTier2(), flow.getMaxLatencyTier2());
+    }
```
Comment (niksv) [inline src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java:476]: I do not see where are you call these 2 methods

Comment (dmitrii-beliakov) [reply src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java]: These methods not used anymore, I moved the logic to `updateRequiredByPathComputationStrategy` and I removed usages, but not the methods themselves. Anyway, now these 2 methods are removed.

File: src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java (hunk 3 above)
Comment (dmitrii-beliakov) [reply src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java]: fixed

**Review by niksv — APPROVED** [review]

**Pull request merged**